src/site/content/en/blog/codelab-payment-form-best-practices/index.md

```diff
@@ -0,0 +1,340 @@
+
+* Does the form work well enough as it is?
+* Is there anything you would change to make it work better?
+* How about on mobile?
+
+Click **View Source** to return to your source code.
+
+## Step 2: Design for mobile and desktop
+
+The HTML you added is valid, but the default browser styling makes the form hard to use, especially 
+on mobile. It doesn't look too good, either.
+
+You need to ensure your forms work well on a range of devices by adjusting padding, margins, and 
+font sizes. 
+
+Copy all the CSS below and paste it into your own `main.css` file.
+
+{% Glitch {
+  id: 'payment-form-codelab-2',
+  path: 'css/main.css',
+  height: 400,
+  previewSize: 0
+} %}
+
+That's a lot of CSS! The main things to be aware of are the changes to sizes:
+
+* `padding` and `margin` are added to inputs.
+* `font-size` and other values are different for different viewport sizes.
+
+When you're ready, click **View App** to see the styled form.  You'll also notice that borders have 
+been adjusted, and `display: block;` is used for labels so they go on a line on their own, and 
+inputs can be full width. [Sign-in form best practices](https://web.dev/sign-in-form-best-practices/#label:~:text=put%20your%20labels%20above%20your%20inputs) 
+explains why this is probably best.
+
+The `:invalid` selector is used to indicate when an input has an invalid value. (You'll use this 
+later in the codelab.)
+
+The CSS is mobile-first:
+
+* The default CSS is for viewports less than `400px` wide.
+* [Media queries](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Media_queries) are 
+used to override the default for viewports that are at least `400px` wide, and then again for 
+viewports that are at least `500px` wide. This should work well for smaller phones, mobile devices 
+with larger screens, and on desktop. 
+
+Whenever you build for the web, you need to test on different devices and viewport sizes. That's 
+especially true for forms, because one small glitch can make them unusable. You should always adjust 
+[CSS breakpoints](/responsive-web-design-basics/#breakpoints) to ensure they work well with your 
+content and your target devices.
+
+* Is the whole form visible? 
+* Are the form inputs big enough? 
+* Is all the text readable?
+* Did you notice any differences between using a real mobile device, and viewing the form in 
+Device Mode in Chrome DevTools?
+* Did you need to adjust breakpoints?
+
+There are several ways to test your form on different devices: {: #different-devices}
+
+* [Use Chrome DevTools Device Mode](https://developers.google.com/web/tools/chrome-devtools/device-mode) 
+to simulate mobile devices. 
+* [Send the URL from your computer to your phone](https://support.google.com/chrome/answer/9430554).
+* Use a service such as [BrowserStack](https://www.browserstack.com/open-source) to test on a range 
+of devices and browsers. 
+
+## Step 3: Add input attributes to enable built-in browser features
```

Comment:
Same as before, let's try to rephrase the heading to be more specific about features.

Reply:
Done.